src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBinderSourceGenerator.Emitter.cs

```diff
@@ -0,0 +1,401 @@
+#pragma warning disable RS1024
+            private readonly HashSet<ITypeSymbol> _bindMethodGen_ProcessedTypes = new(SymbolEqualityComparer.Default);
+            private readonly HashSet<ITypeSymbol> _unsupportedTypes = new(SymbolEqualityComparer.Default);
+#pragma warning restore RS1024
+
+            private readonly INamedTypeSymbol? _typeSymbolForIConfigurationSection;
+
+            // Collections
+            private readonly INamedTypeSymbol _typeSymbolForIEnumerable;
+            private readonly INamedTypeSymbol? _typeSymbolForGenericIEnumerable;
+
+            private readonly INamedTypeSymbol? _typeSymbolForGenericIDictionary;
+            private readonly INamedTypeSymbol? _typeSymbolForGenericDictionary;
+
+            // Postfix for stringValueX variables used to save config value indexer
+            // results e.g. if (configuration["Key"] is string stringValue0) { ... }
+            private int _stringValueCount;
+
+            private readonly SourceWriter _writer = new();
+
+            public Emitter(
+                ConfigurationBinderSourceGenerationContext sourceGenerationContext, Parser.InputTypes targetTypes)
+            {
+                _sourceGenerationContext = sourceGenerationContext;
+                _targetTypes = targetTypes;
+
+                Compilation compilation = _sourceGenerationContext.Compilation;
+                _typeSymbolForIConfigurationSection = compilation.GetBestTypeByMetadataName(FullNameForIConfigurationSectionType);
+                _typeSymbolForIEnumerable = compilation.GetSpecialType(SpecialType.System_Collections_IEnumerable);
+                _typeSymbolForGenericIEnumerable = compilation.GetSpecialType(SpecialType.System_Collections_Generic_IEnumerable_T)?.ConstructUnboundGenericType();
+                _typeSymbolForGenericIDictionary = compilation.GetBestTypeByMetadataName(FullNameForGenericIDictionary)?.ConstructUnboundGenericType();
+                _typeSymbolForGenericDictionary = compilation.GetBestTypeByMetadataName(FullNameForGenericDictionary)?.ConstructUnboundGenericType();
+            }
+
+            public void Emit()
+            {
+                const string LiteralFor_GeneratedConfigurationBinder = "GeneratedConfigurationBinder";
+
+                _writer.WriteSource(@"// <auto-generated/>
+#nullable enable annotations
+#nullable disable warnings
+");
+
+                _writer.WriteBlockStart($"public static class {LiteralFor_GeneratedConfigurationBinder}");
+                EmitConfigureMethod();
+                EmitGetMethod();
+                EmitBindMethods();
+                _writer.WriteBlockEnd();
+
+                SourceText source = SourceText.From(_writer.GetSource(), Encoding.UTF8);
+                _sourceGenerationContext.AddSource($"{LiteralFor_GeneratedConfigurationBinder}.g.cs", source);
+            }
+
+            private void EmitConfigureMethod()
```

Comment:
>EmitConfigureMethod [](http://example.com/codeflow?start=25&length=19)

maybe will be helpful if you write a comments for the Emit methods containing example of the generated code from each method.